src/ZkFold/Base/Algebra/EllipticCurve/Class.hs

```diff
@@ -40,6 +47,14 @@ instance (EllipticCurve curve, Eq (BaseField curve)) => Eq (Point curve) where
 instance EllipticCurve curve => AdditiveSemigroup (Point curve) where
     (+) = add
 
+instance {-# OVERLAPPABLE #-}
+    ( EllipticCurve curve
+    , Eq s
+    , BinaryExpansion s
+    , Bits s ~ [s]
+    ) => Scale s (Point curve) where
```

Comment:
In principle, this instance does the same thing that `Scale Natural` does. Maybe we can replace it with a single
```haskell
instance (EllipticCurve curve, ToConstant s, Const s ~ Natural) => Scale s (Point curve) where
  scale = natScale . toConstant
```
?

Reply:
(Though this is not really lawful because we also need `s` to have appropriate order in case it is finite...)

Reply:
Though this is a topic for another PR as this would invlove changing some of the elliptic curve code, I guess